### PR TITLE
feat(issues): Add scrollCarousel component

### PR DESCRIPTION
**Pull request by scttcper**
Adds a ScrollCarousel component that is very similar to the existing Carousel (added in https://github.com/getsentry/sentry/pull/54433) with some key differences:
- Browser scrollbars are hidden
- There's a white mask indicating there is more to scroll
- The buttons are different
- It jumps more than one item left/right on arrow click

https://github.com/user-attachments/assets/10b4916d-0fa0-4f2b-bd09-a52148724015

what it would look like on issue details

https://github.com/user-attachments/assets/82ea7da8-d647-46b6-8006-920ecd28c1d7

scroll on issue tracking
image

### CODE REVIEW

File: static/app/components/group/externalIssuesList/streamlinedExternalIssueList.tsx [hunk 1]
```diff
@@ -78,8 +79,9 @@ export function StreamlinedExternalIssueList({
               </Tooltip>
             </ErrorBoundary>
           ))}
-          {integrations.length
-            ? integrations.map(integration => {
+          {integrations.length ? (
+            <ScrollCarousel>
```
Comment (malwilley) [inline static/app/components/group/externalIssuesList/streamlinedExternalIssueList.tsx:83]: Are you sure you want to use the carousel for this? IMO the buttons seem better wrapping to multiple lines

File: static/app/utils/useRefChildrenVisibility.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  visibleRatio,
+}: UseRefChildrenVisibilityOptions) {
+  // The visibility match up to the elements list. Visibility of elements is
+  // true if visible in the scroll container, false if outside.
+  const [childrenEls, setChildrenEls] = useState<HTMLElement[]>([]);
+  const [visibility, setVisibility] = useState<boolean[]>([]);
+
+  // Update list of children element
+  useEffect(
+    () => setChildrenEls(Array.from(ref.current?.children ?? []) as HTMLElement[]),
```
Comment (malwilley) [inline static/app/utils/useRefChildrenVisibility.tsx:22]: Seems like this component won't work if the children change after first render, is that right?

File: static/app/components/scrollCarousel.tsx [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    </ScrollCarouselWrapper>
+  );
+}
+
+const ScrollCarouselWrapper = styled('div')`
+  position: relative;
+  overflow: hidden;
+`;
+
+const ScrollContainer = styled('div')`
```
Comment (malwilley) [inline static/app/components/scrollCarousel.tsx:101]: Feels pretty good as is, but do you think it's worth adding `scroll-snap-type`?

File: static/app/utils/useRefChildrenVisibility.tsx (hunk 2 above)
Comment (scttcper) [reply static/app/utils/useRefChildrenVisibility.tsx]: added back children

**Review by malwilley — APPROVED** [review]

File: static/app/components/scrollCarousel.tsx (hunk 3 above)
Comment (scttcper) [reply static/app/components/scrollCarousel.tsx:111]: i'm still playing around with that

**Comment by malwilley** [comment]
@scttcper the arrow button seems a bit small and kinda ovular?

image

File: static/app/components/scrollCarousel.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+import storyBook from 'sentry/stories/storyBook';
+import {space} from 'sentry/styles/space';
+
+import {ScrollCarousel} from './scrollCarousel';
+
+export default storyBook(ScrollCarousel, story => {
+  story('Default', () => (
+    <Fragment>
+      <p>
+        <JSXNode name="SegmentedControl" /> will detect if the content overflows and show
```
Comment (malwilley) [inline static/app/components/scrollCarousel.stories.tsx:15]: Wrong name!

**Pull request merged**